sympy/assumptions/cnf.py

```diff
@@ -262,6 +276,31 @@ def _not(self):
             ll = ll._or(CNF(p))
         return ll
 
+    def rcall(self, expr):
+        from sympy.assumptions.sathandlers import UnevaluatedOnFree, CustomLambda
+        from sympy.assumptions.ask import Predicate
+        clause_list = list()
+        for clause in self.clauses:
+            lits = list()
+            for arg in clause:
+                if isinstance(arg.lit, Predicate):
+                    lits.append(Literal(arg.lit(expr), arg.is_Not))
```

Comment:
Instead of an isinstance chain here can we have a private method on the various classes to implement this.